pkg/manifests/config.go

```diff
@@ -446,6 +427,35 @@ func (c *Config) LoadEnforcedBodySizeLimit(pcr PodCapacityReader, ctx context.Co
 	return poperator.ValidateSizeField(c.ClusterMonitoringConfiguration.PrometheusK8sConfig.EnforcedBodySizeLimit)
 }
 
+func (c *Config) precheck() error {
+	if c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile != FullCollectionProfile && !c.TechPreview {
+		return errors.Wrap(ErrConfigValidation, "collectionProfiles is a TechPreview feature, to be able to use a profile different from the default (\"full\") please enable TechPreview")
+	}
+
+	// Validate the configured collection profile iff tech preview is enabled, even if the default profile is set.
+	if c.TechPreview {
+		for _, profile := range SupportedCollectionProfiles {
+			var v float64
+			if profile == c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile {
+				v = 1
+			}
+			metrics.CollectionProfile.WithLabelValues(string(profile)).Set(v)
+		}
+		if !slices.Contains(SupportedCollectionProfiles, c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile) {
+			return errors.Wrap(ErrConfigValidation, fmt.Sprintf(`%q is not supported, supported collection profiles are: %q`, c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile, SupportedCollectionProfiles.String()))
+		}
+	}
+
+	var d float64
+	// Highlight deprecated config fields.
+	if c.ClusterMonitoringConfiguration.K8sPrometheusAdapter.DedicatedServiceMonitors != nil {
```

Comment:
This will not cover:
```
k8sPrometheusAdapter:
  dedicatedServiceMonitors:
``` 
(where `dedicatedServiceMonitors` is empty)
I'll see if we could cover it as well

Reply:
I don't think it's easy to catch that without custom unmarshalling which is not worth it...

Reply:
I think that's fine. We want to alert users when they use a deprecated field, even if empty.

Reply:
Actually both
```
k8sPrometheusAdapter:
  dedicatedServiceMonitors:
```
and
```
k8sPrometheusAdapter:
```

result in `K8sPrometheusAdapter.DedicatedServiceMonitors=nil`, so we'll not be able to warn on empty `dedicatedServiceMonitors`.